### DISCUSSION TITLE
GeoServer - Änderung der Länge des escape char ab Version 2.22

### DISCUSSION THREAD

**Discussion by faulgurke** [e-netze/webgis-community#38] [General]
Ab Version 2.22 darf ein an GeoServer übergebener escape Character nur noch 1 Zeichen lang sein. 
https://github.com/geotools/geotools/blob/main/modules/library/main/src/main/java/org/geotools/filter/LikeFilterImpl.java

![esc_char](https://user-images.githubusercontent.com/100088983/217459313-84a45d7b-9a3d-40cb-929d-5148f25e1b12.png)

Dies hat Auswirkungen auf Wildcard-Suchen in GeoServer (WFS).

![esc_char2](https://user-images.githubusercontent.com/100088983/217459744-979594af-46a2-4dc2-b224-5296f1ed612d.png)
Sobald bei uns die Version 2.22 verfügbar ist, können wir ein Beispiel liefern.

**Comment by gis-eni** [comment]
Gehört das dann nicht ins GeoServer Forum? ;)

**Comment by faulgurke** [comment]
Das gehörte dann ins GeoServer Forum wenn nicht auch WebGIS bei der Verwendung von WFS Services betroffen wäre.
Da es aber so ist (und die Kollegen Wasserwirtschaft das auch schon bemerkt haben), wollte ich rechtzeitig darauf hinweisen.

**Comment by gis-eni** [comment]
Na gut, ich versteh bisher noch nicht, was das bedeutet, aber wenn sich dadurch der WFS Standard ändert (sprich für alle WFS Server) werden wir die notwendigen Änderungen nachziehen. Wenn das irgendeine GeoServer Spezialität ist, frage mich mich schon, ob es sich hier eher um einen BUG handelt?!
